File: BookingApp/frontend/src/app/shared/models/LoyaltyProgram.ts
export class LoyaltyProgram{
    public id : Number;
    public pointsPerReservation: Number;
    public silverMemberThreshold: Number;
    public goldenMemberThreshold: Number;

    constructor(id : Number,pointsPerRes : Number,silverMemberThreshold : Number,goldMemberThreshold : Number){
        this.id = id;
        this.pointsPerReservation = pointsPerRes;
        this.silverMemberThreshold = silverMemberThreshold;
        this.goldenMemberThreshold = goldMemberThreshold;
    }
}

File: BookingApp/src/main/java/com/example/BookingApp/users/model/LoyaltyProgram.java
package com.example.BookingApp.users.model;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class LoyaltyProgram {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator="seq")
    @GenericGenerator(name = "seq", strategy="increment")
    private Long id;
    private double pointsPerReservation;
    private double silverMemberThreshold;
    private double goldenMemberThreshold;
}
